### DISCUSSION TITLE
MSSQL Limit Issue

### DISCUSSION THREAD

**Discussion by cbbalanquit** [getredash/redash#6550] [Q&A] [answered]
When trying to query MSSQL, I guess it still use LIMIT 1000 as default addition to the query. It was resolved from this issue ([issue5528](https://github.com/getredash/redash/issues/5528)) but not sure why it still persist now even my redash instance is the updated one right now

![Screenshot from 2023-10-24 22-04-20](https://github.com/getredash/redash/assets/81967711/b4886766-bb14-4890-b50a-059cfd947798)

**Comment by justinclift** [comment, marked as answer]
Yeah, I've seen this happen with a customer's production instance as well, several months ago.

The solution in that case was to patch the source, so that the 1000 limit row wasn't applied for any data source.  That's fine for this particular customer, but obviously isn't what's needed for the "full  and proper solution" for everyone.

**Comment by cbbalanquit** [comment]
I see. What part in the codebase should I modify to path this 1000 limit? not really familiar yet in where to look

**Reply by cbbalanquit** [reply]
I think I saw it. changed the default to false

![Screenshot from 2023-10-25 21-11-49](https://github.com/getredash/redash/assets/81967711/3471b9ea-b4f6-43ef-bfc6-4a727e7c8502)

**Reply by justinclift** [reply]
Screenshot seems busted?

**Reply by cbbalanquit** [reply]
I reupload it

**Reply by justinclift** [reply]
Cool.  When I did it, I just ripped out the Python backend code that implemented the query limit.

But it looks like your approach might work too.

Does it seem to? :smile:

**Comment by cbbalanquit** [comment]
Yes it worked too, luckily. I can already query without error in
limits..but not sure tho how can i adjust this for mssql only so once i
deployed in prod

**Reply by justinclift** [reply]
Cool.  Good that it worked.

But yeah, I'm not sure how to do the "proper" implementation either (yet).
